### PR TITLE
Documentation improvements; add UnliftIO.Concurrent module.

### PR DESCRIPTION
**Pull request by nickkuk**

### CODE REVIEW

File: unliftio/ChangeLog.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
-## 0.1.1.0
+## 0.2.0.0
```
Comment (snoyberg) [inline unliftio/ChangeLog.md]: Adding a new module isn't a breaking change. Could you make the version `0.1.2.0` instead?

File: unliftio/src/UnliftIO/Concurrent.hs [hunk 2]
```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE RankNTypes #-}
+-- | Unlifted "Control.Concurrent".
+--
+-- Use this module only if "UnliftIO.Async" is not enough.
```
Comment (snoyberg) [inline unliftio/src/UnliftIO/Concurrent.hs]: Thank you for including this! Perhaps it's also worth nothing that, unlike most other modules in the package, this one is not reexported by `UnliftIO`.

**Comment by nickkuk** [comment]
I just left 0.1.1.0, as it didn't appear on Stackage. Is it ok?

What do you think about
```haskell
killThread tid = throwTo tid ThreadKilled
```
vs
```haskell
killThread = liftIO . C.killThread
```
?

**Comment by snoyberg** [comment]
Oh, good call on the version number, thanks.

I think I'd tend towards the latter, what's the advantage of the former?

**Comment by nickkuk** [comment]
In the lastest version I use `killThread = liftIO . C.killThread`.
Sorry for the noise.

**Comment by snoyberg** [comment]
No noise at all, I appreciate the work here! I'll give one more review, and once Travis passes merge and release. Thanks!

**Pull request merged**